pkg/clusterid/cluster_id_test.go

```diff
@@ -0,0 +1,44 @@
+package clusterid_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/clusterid"
+	kuma_cp "github.com/kumahq/kuma/pkg/config/app/kuma-cp"
+	"github.com/kumahq/kuma/pkg/test/runtime"
+)
+
+var _ = Describe("Cluster ID", func() {
+
+	var stop chan struct{}
+
+	AfterEach(func() {
+		if stop != nil {
+			close(stop)
+		}
+	})
+
+	It("should create and set cluster ID", func() {
+		// given runtime with cluster ID components
+		cfg := kuma_cp.DefaultConfig()
+		builder, err := runtime.BuilderFor(cfg)
+		Expect(err).ToNot(HaveOccurred())
+		runtime, err := builder.Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = clusterid.Setup(runtime)
+		Expect(err).ToNot(HaveOccurred())
+
+		// when runtime is started
+		stop = make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			err := runtime.Start(stop)
+			Expect(err).ToNot(HaveOccurred())
+		}()
+
+		// then cluster ID is created and set in Runtime object
+		Eventually(runtime.GetClusterId, "5s", "100ms").ShouldNot(BeEmpty())
```

Comment:
I know it was the same before your changes, but is it okay that we can dispatch empty `cluster_id`?

Reply:
yes, it is ok.